porch/controllers/workloadidentitybinding/pkg/controllers/workloadidentitybinding/controller.go

```diff
@@ -211,9 +230,55 @@ func (r *WorkloadIdentityBindingReconciler) BuildObjectsToApply(ctx context.Cont
 		})
 		obj.SetOwnerReferences(ownerRefs)
 	}
+
 	return objects, nil
 }
 
+func (r *WorkloadIdentityBindingReconciler) addWiAnnotation(ctx context.Context, projectID string, subject *api.WorkloadIdentityBinding) error {
+	sa, err := r.getServiceAccount(ctx, subject)
+	if err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	gsaEmail := fmt.Sprintf("%s@%s.iam.gserviceaccount.com", subject.Spec.ResourceRef.Name, projectID)
+
+	patch := client.MergeFrom(sa.DeepCopy())
```

Comment:
My understanding is that we should not do this now that we have server-side apply.

My understanding is that we should construct an object with only the fields we care about, either by using unstructured.Unstructured (my preference) or the new "for-apply" variants of the typed objects.

Reply:
Updated this to use the SSA patch with unstructured.